src/mapDataManipulation/mapDataManipulationUtils.ts

```diff
@@ -1,22 +1,59 @@
-import { RgbMappingArrays, Effects, ColorRange } from './const';
+import { Effects, ColorRange, ImageProperties } from './const';
 
-export function prepareRgbMappingArrays(): RgbMappingArrays {
-  return {
-    red: [...Array(256).keys()],
-    green: [...Array(256).keys()],
-    blue: [...Array(256).keys()],
-  };
+export async function getImageData(originalBlob: Blob): Promise<ImageProperties> {
```

Comment:
These functions seem to be very generic. I think it would be a good idea to put them next to [existing utility functions](https://github.com/sentinel-hub/sentinelhub-js#utility-functions) to avoid duplication of efforts? Exposing them might not be needed at this time, this would serve just to avoid ending up with multiple instances of the same code in many places (please make sure that the code is not duplicated elsewhere).

Reply:
I moved them to `/utils/canvas` as suggested and used `canvasToBlob` in `getBlob`. There is unfortunately some duplicated code as described below.